x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -333,14 +333,30 @@ class AgentPolicyService {
       withPackagePolicies = false,
     } = options;
 
-    const agentPoliciesSO = await soClient.find<AgentPolicySOAttributes>({
+    const baseFindParams = {
       type: SAVED_OBJECT_TYPE,
       sortField,
       sortOrder,
       page,
       perPage,
-      filter: kuery ? normalizeKuery(SAVED_OBJECT_TYPE, kuery) : undefined,
-    });
+    };
+    const filter = kuery ? normalizeKuery(SAVED_OBJECT_TYPE, kuery) : undefined;
+    let agentPoliciesSO;
+    try {
+      agentPoliciesSO = await soClient.find<AgentPolicySOAttributes>({ ...baseFindParams, filter });
+    } catch (e) {
+      const isBadRequest = e.output?.statusCode === 400;
+      const isKQLSyntaxError = e.message?.startsWith('KQLSyntaxError');
```

Comment:
I wouldve liked this check to be more robust but there is nothing much more I can use to distinguish KQL errors, KQL error:

```
{
    "isBoom": true,
    "isServer": false,
    "data": null,
    "output": {
        "statusCode": 400,
        "payload": {
            "message": "KQLSyntaxError: Expected \"(\", \"{\", value, whitespace but end of input found.\ningest-agent-policies.attributes.description :    \n--------------------------------------------------^: Bad Request",
            "statusCode": 400,
            "error": "Bad Request"
        },
        "headers": {}
    }
}
```
VS "yo this isn't even KQL" error:

```
{
    "isBoom": true,
    "isServer": false,
    "data": null,
    "output": {
        "statusCode": 400,
        "payload": {
            "message": "The key is empty and needs to be wrapped by a saved object type like ingest-agent-policies: Bad Request",
            "statusCode": 400,
            "error": "Bad Request"
        },
        "headers": {}
    }
}
```

Reply:
Yes I think we do not have better alternative than looking for text here, hopefully this does not change :)